pgmpy/factors/discrete/DiscreteFactor.py

```diff
@@ -710,6 +710,11 @@ def copy(self):
         # because __init__ methods does that.
         return DiscreteFactor(self.scope(), self.cardinality, self.values)
 
+    def is_valid_cpd(self):
+        return np.allclose(self.to_factor().marginalize(self.scope()[:1], inplace=False).values.flatten('C'),
+                            np.ones(np.product(self.cardinality[:0:-1])),
+                            atol=0.01)
```

Comment:
It was moved from [this line](https://github.com/pgmpy/pgmpy/blame/dev/pgmpy/models/BayesianModel.py#L360) which was introduced by @abinashpanda. I think @abinashpanda is the best person to answer this question.